core/admin/server.go

```diff
@@ -40,7 +40,7 @@ type Server struct {
 	srv *http.Server
 
 	nodeWallet             *NodeWallet
-	protocolUpgradeService ProtocolUpgradeAdminService
+	protocolUpgradeService *ProtocolUpgradeAdminService
```

Comment:
why a pointer to an interface here?

Reply:
oh, I see it's not the interface it's the struct, ok makes sense.

Reply:
It's not interface. It's the service stuct. And it has to be a pointer otherwise it will not get registered as service with the `RegisterService` RPC server method.